v3 - Add formatting checks to Python linting process

alexandreNoens opened the pull request:
Ensure `make lint-python` also checks code formatting, not just lint rules. This helps maintain consistent style across files and avoids formatting discrepancies.

Resolves: #23

alexandreNoens committed:
Add formatting checks to Python linting process

Ensure `make lint-python` also checks code formatting, not just lint
rules. This helps maintain consistent style across files and avoids
formatting discrepancies.

Resolves: #23

gentooboontoo commented:
Mob : à tester @gentooboontoo @SajedehAdelia

gentooboontoo reviewed (approved):
It works in principle :+1:

Small note: pellov overrides make lint-python in the Makefile (otherwise djlint crashes), so it has no effect. I think this override should be removed and the absence of Jinja2 templates should be handled better. I’ll take a look to see if I can fix it, otherwise I’ll create a ticket for later.

gentooboontoo commented:
Suggestions to improve this Python linting target in another commit:

- check if `TEMPLATE_DIRS` exists 
- `--no-run-if-empty` added as argument to `xargs` to prevent failure when there is no template files (empty list)

```diff
diff --git a/base.Makefile b/base.Makefile
index fe77d62..c7dbeff 100644
--- a/base.Makefile
+++ b/base.Makefile
@@ -326,7 +326,10 @@ lint-pytho%: ## lint-python: Lint python source
        $(LOG)
        $(RUFF) check $(LINTED_PYTHON_DIRS)
        $(RUFF) format --diff $(LINTED_PYTHON_DIRS)
-       find $(TEMPLATE_DIRS) -type f -name '*.jinja2' | xargs $(DJLINT) --check --lint
+       if [ -d $(TEMPLATE_DIRS) ]; then \
+               find $(TEMPLATE_DIRS) -type f -name '*.jinja2' | \
+               xargs --no-run-if-empty $(DJLINT) --check --lint; \
+       fi

 lint-nod%: ## lint-node: Lint node source
        $(LOG)
```

alexandreNoens committed:
Add a condition for the fix and lint Python commands

Some projects don't have a Jinja2 template, so there's no need to use
djLint to format them.

SajedehAdelia reviewed (commented):
:eyes:

alexandreNoens committed:
Improve the printout

Now it displays the executed command.
And I used a variable to store the djlint command.

alexandreNoens commented:
> Suggestions to improve this Python linting target in another commit:

I added a print to see what the command does.
This way, the logs we see are the same as before, with the condition added.

I have some doubts about the name of the variable that contains the command.

If it's okay with you, I'll add this commit to the v5 branch.
@gentooboontoo

gentooboontoo commented on base.Makefile:
```diff
@@ -46,6 +46,7 @@ TEMPLATE_DIRS ?= $(PYTHON_SRCDIR)/templates
 PYTHON_PKG_TOOLS ?= pip pip-tools setuptools wheel
 ### Commands (from `PYTHON_BINDIR` via `PATH` environment variable)
 DJLINT ?= djlint
+DJLINT_JINJA_CMD = find $(TEMPLATE_DIRS) -type f -name '*.jinja2' | xargs --no-run-if-empty $(DJLINT)
```
Use [simply expanded assignment](https://www.gnu.org/software/make/manual/html_node/Simple-Assignment.html) here (`:=`). No need for recursive assignment, it should only be expanded once.

gentooboontoo reviewed (approved):
I did not check but it looks fine. Just change the assignment type. I pre-approve.

alexandreNoens committed:
Fix assignment operator

This ensures the command is expanded once at definition time,
which avoids unnecessary repeated evaluations.
